File: common/python_common/mongo_manipulate.py
# -*- coding: utf-8 -*-
#
#	mongo_manipulate.py
#
#					Aug/05/2018
#
import	pymongo
import	datetime
#
from text_manipulate import dict_append_proc
# -------------------------------------------------------------------
def	mongo_update_proc (db_aa,collection,key_in,population_in):
	date_mod = datetime.datetime.now ()
	db_aa[collection].update({"key": key_in},{"$set":{"population": population_in,"date_mod": '%s' % date_mod}})
#
# -------------------------------------------------------------------
def	mongo_delete_proc (db_aa,collection,key_in):
	db_aa[collection].remove ({"key": key_in})
#
# -------------------------------------------------------------------
def	mongo_to_dict_proc (db_aa,collection):
#
	dict_aa = {}

	for item in db_aa[collection].find().sort("key", pymongo.ASCENDING):
		dict_aa = dict_append_proc (dict_aa,item["key"],item["name"],item["population"],item["date_mod"])
#
	return	dict_aa
#
# -------------------------------------------------------------------
def dict_to_mongo_proc (db_aa,collection,dict_aa):
#
	db_aa[collection]
	db_aa[collection].remove ()
#
	for key in dict_aa.keys():
		unit = dict_aa[key]
		db_aa[collection].save({"key": key,"name": unit['name'],"population": unit['population'],"date_mod": unit['date_mod']})
#
# -------------------------------------------------------------------

File: common/php_common/mongo_manipulate.php
<?php
// --------------------------------------------------------------------
/*
	mongo_manipulate.php

					Aug/01/2018

*/
// --------------------------------------------------------------------
function mongo_to_dict_proc ($user,$password)
{
	$dict_aa = array ();

	$str_connect = "mongodb://" . $user . ":" . $password . "@localhost:27017";
	$manager = new MongoDB\Driver\Manager($str_connect);

	$filter = [];
	$options = [
		'projection' => ['_id' => 0],
		'sort' => ['_id' => -1],
	];

	$query = new MongoDB\Driver\Query($filter, $options);
	$cursor = $manager->executeQuery('city.saitama', $query);

	$dict_aa = array ();

	foreach ($cursor as $document) {
		$dict_aa = dict_append_proc ($dict_aa,$document->key,$document->name,$document->population,$document->date_mod);
	}

	return	$dict_aa;
}

// --------------------------------------------------------------------
function mongo_update_proc ($id_in,$population_in,$user,$password)
{
	$str_connect = "mongodb://" . $user . ":" . $password . "@localhost:27017";
	$manager = new MongoDB\Driver\Manager($str_connect);

	$bulk = new MongoDB\Driver\BulkWrite;

	date_default_timezone_set('Asia/Tokyo');
	$today = date ("Y-m-d");

	$newdata = array('$set' => array("population" => $population_in,"date_mod" => $today));

	$bulk->update(array("key" => $id_in), $newdata);

	$manager->executeBulkWrite('city.saitama', $bulk);
}

// --------------------------------------------------------------------
function mongo_delete_proc ($id_in,$user,$password)
{
	$str_connect = "mongodb://" . $user . ":" . $password . "@localhost:27017";
	$manager = new MongoDB\Driver\Manager($str_connect);

	$bulk = new MongoDB\Driver\BulkWrite;

	$bulk->delete(['key' => $id_in]);

	$result = $manager->executeBulkWrite('city.saitama', $bulk);
}

// --------------------------------------------------------------------
function dict_to_mongo_proc ($dict_aa,$user,$password)
{
	$str_connect = "mongodb://" . $user . ":" . $password . "@localhost:27017";

	$manager = new MongoDB\Driver\Manager($str_connect);

	$manager->executeCommand('city', new \MongoDB\Driver\Command(["drop" => "saitama"]));


	$bulk = new MongoDB\Driver\BulkWrite;
	foreach ($dict_aa as $key => $value)
		{
	$bulk->insert(['key' => $key, 'name' => $value["name"], 'population' => $value['population'], 'date_mod' => $value['date_mod']]);
		}

	$manager->executeBulkWrite('city.saitama', $bulk);
}

// --------------------------------------------------------------------
?>
